### PR TITLE
HEEDLS-866 Refactor reports page when there is no activity at the centre

### PR DESCRIPTION
**Pull request by VladNicol**
### JIRA link
[HEEDLS-866](https://softwiretech.atlassian.net/jira/software/c/projects/HEEDLS/boards/753?modal=detail&selectedIssue=HEEDLS-866)

### Description
On the reports page, if there was no activity at the centre, the query inside the `ActivityDataService.GetStartOfActivityForCentre` method was returning null. This caused the change filters and download buttons to give an error. I added a check and if there is no activity at the centre/for the course category that the current user has access to, I don't display the buttons and instead show an informative message.

### Screenshots
No activity for all course categories:
image

No activity for the specific course category the user has access to:
image

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Data/DataServices/ActivityDataService.cs [hunk 1]
```diff
@@ -72,14 +72,21 @@ SELECT ap.ApplicationID
...
-                new { centreId }
-            );
+                    new { centreId }
+                )
+                : connection.QuerySingleOrDefault<DateTime?>(
+                    @"SELECT MIN(LogDate)
+                    FROM tActivityLog
+                    WHERE CentreID = @centreId AND CourseCategoryId = @courseCategoryId",
+                    new { centreId, courseCategoryId }
+                );
```
Comment (AlexJacksonDS) [inline DigitalLearningSolutions.Data/DataServices/ActivityDataService.cs:89]: This stuff doesn't need to be 2 separate SQL queries. The pattern we've been using for category ids is `(CourseCategoryID = @courseCategoryId OR @courseCategoryId IS NULL)`

**Review by AlexJacksonDS — COMMENTED** [review]
Just one comment about the SQL query

File: DigitalLearningSolutions.Data/DataServices/ActivityDataService.cs (hunk 1 above)
Comment (VladNicol) [reply DigitalLearningSolutions.Data/DataServices/ActivityDataService.cs:89]: That makes sense, thank you!

**Review by GeorgeBlackwell — APPROVED** [review]
Nice work, nothing to add beyond Alex's comment which you've tackled.

**Pull request merged**